x-pack/plugin/identity-provider/src/internalClusterTest/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderIndexTests.java

```diff
@@ -264,4 +224,20 @@ private static <T> ActionListener<T> assertListenerIsOnlyCalledOnce(ActionListen
         });
     }
 
+    // Since we don't actually enable the IDP plugin, it doesn't initialize its template registry, so
+    // this creates just the template registry on its own.
```

Comment:
I find it super weird that our integration test for the plugin does not enable the plugin :/

I assume this is here to allow the other tests to pass ? 

No issues if that is case - but it would be nice to also include a quick test in IdpRestTestCase ( a real REST test with the plugin enabled) that you can read the index template and the versions are as expected.

Reply:
This isn't the only integration test for this plugin - there's a [whole base class](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/identity-provider/src/internalClusterTest/java/org/elasticsearch/xpack/idp/saml/test/IdentityProviderIntegTestCase.java) to do the involved setup necessary required to fully enable the plugin for a full `IntegTest`. This test plugin is only used in this test suite, specifically testing template behavior - I believe the other integration tests passing relies on the template being in place, and get that from the normal flow.

I'll clarify this comment.